### PR TITLE
Create composer.json

### PR DESCRIPTION
**Pull request by liKe2k1**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "name": "like2k1/gentelella",
```
Comment (christianesperar) [inline composer.json]: I think it's better if puikinsh/gentelella

Comment (liKe2k1) [inline composer.json]: Ops, sorry internal testing trash.. Fixed

**Comment by krzysiekpiasecki** [comment]
Composer is Dependency Manager for PHP. It's not a php project. Managing assets e.g via bower.

**Comment by puikinsh** [comment]
I think that the idea is to help PHP developers to use Gentelella as part of their PHP based projects. Am I right @liKe2k1 ?

**Comment by krzysiekpiasecki** [comment]
What are the benefits?

IMHO

I'm sure that not every front-end vendor is registred in the packagist, so anyway we MUST use bower to manage assets. In my practice I always manage via composer only with php dependencies. What are the benefits for php developers?

**Comment by liKe2k1** [comment]
You,re right. You can integrate gentelella very easy with composer in new or existing php project with required (like bootstrap) or optional dependencies.


In a PHP Environment, composer is the state-of-the-art package manager. 


Outlook for Android herunterladen



Von: Aigars Silkalns

Gesendet: Mittwoch, 18. Mai, 19:35

Betreff: Re: [puikinsh/gentelella] Create composer.json (#155)

An: puikinsh/gentelella

Cc: Tobias Ehrig, Mention



I think that the idea is to help PHP developers to use Gentelella as part of their PHP based projects. Am I right @liKe2k1 ?

**Comment by krzysiekpiasecki** [comment]
@liKe2k1 

> Afaik bower is only usefully if you are developing an front-end only application and have no dependencies to other languages than js.

This repository is just a front-end application and composer is out of the scope of this repository.

 Creating project from a composer is simply just a cloning repository and then installing vendors.
- The structure of repository is 'useless' for php projects. The files are static html with duplication.
- Gentelella vendors must be moved from vendors to public 'web' location - where is this location for my concrete project?
- You must still include vendors manually in your source code (php has an autoloading),
- ...

It's not an easy work and *you never prepare distribution file for php devs*. What are concrete benefits? What are the PHP projects that fully manage assets with the composer?


BTW Bower is much faster and reliable and has a great community experience.

**Comment by fps01** [comment]
@liKe2k1 
For using this theme in php you can add a `fxp/composer-asset-plugin` package (https://github.com/francoispluchino/composer-asset-plugin) to `composer.json` of your project.
I did it for an extension for yii2. You can see example here (https://github.com/yiister/yii2-gentelella/blob/master/composer.json).

**Comment by krzysiekpiasecki** [comment]
@liKe2k1 or if you want you can use build tools like phing to let the composer be a php dependency manager.

**Comment by christianesperar** [comment]
Please remove the dependencies since this is already on the vendor folder

**Pull request closed without merging**